deckhouse-controller/pkg/controller/module-controllers/release/controller.go

```diff
@@ -411,10 +418,19 @@ func (c *moduleReleaseReconciler) reconcilePendingRelease(ctx context.Context, m
 			return ctrl.Result{RequeueAfter: defaultCheckInterval}, nil
 		}
 	} else {
-		if e := c.updateModuleReleaseStatusMessage(ctx, mr, fmt.Sprintf("Update policy not set. Create a ModuleUpdatePolicy object and label the release '%s=<policy_name>'", UpdatePolicyLabel)); e != nil {
-			return ctrl.Result{Requeue: true}, e
+		// get all policies regardless of their labels
+		var policies = new(v1alpha1.ModuleUpdatePolicyList)
+		err = c.client.List(ctx, policies)
+		if err != nil {
+			return ctrl.Result{Requeue: true}, err
+		}
+		policy, err = c.getReleasePolicy(mr.GetModuleSource(), mr.GetModuleName(), policies.Items)
```

Comment:
We have to set found policy for ModuleRelease. Otherwise it works without found policy

Reply:
```
# kubectl get mr
NAME                        PHASE        UPDATE POLICY    TRANSITIONTIME   MESSAGE
test-v0.8.23                Deployed     test-alpha       3h17m
test-v0.9.0                 Pending                       22s              Waiting for manual approval
```

Reply:
I created a configuration:
```yaml
---
apiVersion: deckhouse.io/v1alpha1
kind: ModuleSource
metadata:
  name: echo
  labels:
    heritage: deckhouse
    module: echo
spec:
  registry:
    ca: ""
    dockerCfg: eyfQ==
    repo: dev-registry.deckhouse.io/deckhouse/redacted/external-modules
    scheme: HTTPS
---
apiVersion: deckhouse.io/v1alpha1
kind: ModuleUpdatePolicy
metadata:
  name: echo
spec:
  moduleReleaseSelector:
    labelSelector:
      matchLabels:
        source: echo
        module: echo
  releaseChannel: Alpha
  update:
    mode: Manual

---
apiVersion: deckhouse.io/v1alpha1
kind: ModuleRelease
metadata:
  annotations:
    dryrun: "true"
  labels:
    module: echo
    source: echo
  name: echo-v0.4.62
spec:
  moduleName: echo
  version: 0.4.62
  weight: 910
```

```sh
# kubectl apply -f echo.yaml
modulesource.deckhouse.io/echo created
moduleupdatepolicy.deckhouse.io/echo created
modulerelease.deckhouse.io/echo-v0.4.62 created

# kubectl get mr
NAME                                     PHASE        UPDATE POLICY     TRANSITIONTIME   MESSAGE
echo-v0.4.62                             Pending      echo              11s              Waiting for manual approval
echo-v0.4.66                             Pending      echo              25s              requirements are not satisfied: current kubernetes version is not suitable: 1.28.12 is less than or equal to v1.30.0
```

As you can see, the update policy is now applied if it was not explicitly set.

Reply:
If I create a release with a non-existent mup:
```yaml
# kubectl get mr
NAME                                     PHASE        UPDATE POLICY     TRANSITIONTIME   MESSAGE
echo-v0.4.62                             Pending      echo              10m              Waiting for manual approval
echo-v0.4.63                             Pending      echo3             7s               Update policy echo3 not found
echo-v0.4.66                             Pending      echo              10m              requirements are not satisfied: current kubernetes version is not suitable: 1.28.12 is less than or equal to v1.30.0
```